grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlerBuilder.java

```diff
@@ -0,0 +1,116 @@
+
+    @Nullable
+    private Set<UnframedGrpcErrorResponseType> responseTypes;
+
+    UnframedGrpcErrorHandlerBuilder(UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        this.statusMappingFunction = statusMappingFunction;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder jsonMarshaller(MessageMarshaller jsonMarshaller) {
+        this.jsonMarshaller = requireNonNull(jsonMarshaller, "jsonMarshaller");
+        return this;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder statusMappingFunction(
+            UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        this.statusMappingFunction = requireNonNull(statusMappingFunction, "statusMappingFunction");
+        return this;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder responseTypes(UnframedGrpcErrorResponseType... responseTypes) {
+        requireNonNull(responseTypes, "responseTypes");
+        if (this.responseTypes == null) {
+            this.responseTypes = EnumSet.noneOf(UnframedGrpcErrorResponseType.class);
+        }
+        this.responseTypes.addAll(ImmutableSet.copyOf(responseTypes));
+        return this;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder registerMarshallers(Message... messages) {
+        requireNonNull(messages, "messages");
+        MessageMarshaller.Builder jsonMarshallerBuilder = jsonMarshaller.toBuilder();
+
+        for (Message message : messages) {
+            jsonMarshallerBuilder = jsonMarshallerBuilder.register(message);
+        }
+
+        jsonMarshaller = jsonMarshallerBuilder.build();
+        return this;
+    }
+
+    @SafeVarargs
+    public final UnframedGrpcErrorHandlerBuilder registerMarshallers(Class<? extends Message>... messageTypes) {
+        requireNonNull(messageTypes, "messageTypes");
+        MessageMarshaller.Builder jsonMarshallerBuilder = jsonMarshaller.toBuilder();
+
+        for (Class<? extends Message> messageType : messageTypes) {
+            jsonMarshallerBuilder = jsonMarshallerBuilder.register(messageType);
+        }
+
+        jsonMarshaller = jsonMarshallerBuilder.build();
+        return this;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder registerMarshallers(Iterable<?> messagesOrMessageTypes) {
+        requireNonNull(messagesOrMessageTypes, "messagesOrMessageTypes");
+        MessageMarshaller.Builder jsonMarshallerBuilder = jsonMarshaller.toBuilder();
+
+        for (final Object messageOrMessageType : messagesOrMessageTypes) {
+            requireNonNull(messageOrMessageType, "messagesOrMessageTypes contains null.");
+            if(messageOrMessageType instanceof Message) {
+                jsonMarshallerBuilder = jsonMarshallerBuilder.register((Message)messageOrMessageType);
+            } else {
+                throw new IllegalArgumentException(messageOrMessageType.getClass().getName() +
+                                                   " is neither Message type nor Message extended type.");
+            }
```

Comment:
We should handle the case where `messageOrMessageType` is a class. We could check like this:

```java
if (messageOrMessageType instanceof Class &&
    Message.class.isAssignableFrom((Class<?>) messageOrMessageType)) { ... }
```